src/commands/title.ts

```diff
@@ -40,30 +56,87 @@ export const command: NeedleCommand = {
 			return interactionReply(interaction, getMessage("ERR_UNKNOWN", interaction.id));
 		}
 
-		const channel = interaction.channel;
-		if (!channel?.isThread()) {
+		const thread = interaction.channel;
+		if (!thread?.isThread()) {
 			return interactionReply(interaction, getMessage("ERR_ONLY_IN_THREAD", interaction.id));
 		}
 
-		const newThreadName = interaction.options.getString("value");
+		let titlePlaceholder = thread.name.replaceAll("🆕", "");
+		if (titlePlaceholder.length > 45) {
+			titlePlaceholder = titlePlaceholder.slice(0, 42) + "...";
+		}
+
+		const modal = new Modal().setCustomId(this.name).setTitle("Set a new thread title");
+
+		const titleInput = new TextInputComponent()
+			.setCustomId("title")
+			.setLabel("Thread title")
+			.setMinLength(1)
+			.setMaxLength(95)
+			.setRequired(true)
+			.setPlaceholder(titlePlaceholder)
+			.setStyle("SHORT");
+
+		const channelId = new TextInputComponent()
+			.setCustomId("threadId")
+			.setLabel("Thread ID (do not change)")
+			.setMinLength(18)
+			.setMaxLength(18)
+			.setRequired(true)
+			.setPlaceholder("Put it back! (Discord bug workaround)")
+			.setStyle("SHORT")
+			.setValue(interaction.channelId);
+
+		const row = new MessageActionRow<ModalActionRowComponent>().addComponents(titleInput);
+		const row2 = new MessageActionRow<ModalActionRowComponent>().addComponents(channelId);
+
+		modal.addComponents(row, row2);
+
+		return interaction.showModal(modal);
+	},
+	async handleModalSubmit(interaction: ModalSubmitInteraction): Promise<void> {
+		const member = interaction.member;
+		if (!(member instanceof GuildMember)) {
+			return interactionReply(interaction, getMessage("ERR_UNKNOWN", interaction.id));
+		}
+
+		let thread = interaction.channel;
+		if (!thread) {
+			return interactionReply(interaction, getMessage("ERR_UNKNOWN", interaction.id));
+		}
+
+		// Very ugly workaround because Discord's modals don't know where they were opened from
```

Comment:
I don't think this is good UX, we should just wait for Discord to fix this issue.

Reply:
I wanted to open a bug report but it's near impossible to understand how to do that, so I have no way of tracking progress of this. If it's a known bug that they talk openly about and give us a good ETA of fixing, I'm down for waiting. But our users have waited a very long time for default titles, and I believe giving them this working (albeit slightly suboptimal UX) version is better than nothing.

Maybe we should revert the changes in this PR like you say since this is mostly me just toying around with having an "Edit title" button - not a super popular request (but it would be improved UX if we didn't have this annoying bug), and instead focus on using modals to finally implement what I want for default titles, the most popular request.

The reason why I wanted to toy around in this PR with the edit title button was just to do something easy get a feel for the API around modals :)

Reply:
You can submit a bug report here 👇

https://support.discord.com/hc/en-us/requests/new?ticket_form_id=360006586013